### PR TITLE
Pass `runtime_mode` gn arg through to `ShouldEnableCheckedMode()`

### PR DESCRIPTION
**Pull request by tvolkert**
https://github.com/flutter/flutter/issues/4866

### CODE REVIEW

**Comment by tvolkert** [comment]
@abarth

File: sky/engine/config.gni [hunk 1]
```diff
@@ -54,18 +55,10 @@ if (sky_asserts_always_on) {
   feature_defines_list += [ "ENABLE_ASSERT=1" ]
 }
 
-if (sky_dart_strict_mode) {
+if (flutter_runtime_mode == "debug") {
   feature_defines_list += [ "ENABLE_DART_STRICT=1" ]
```
Comment (abarth) [inline sky/engine/config.gni]: I'd remove ENABLE_DART_STRICT and trigger it off FLUTTER_RUNTIME_MODE

**Comment by abarth** [comment]
LGTM

File: sky/engine/config.gni (hunk 1 above)
Comment (tvolkert) [inline sky/engine/config.gni]: Done

**Pull request merged**